WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationCommentDetailViewController.swift

```diff
@@ -156,21 +164,41 @@ private extension NotificationCommentDetailViewController {
             return
         }
 
-        if commentDetailViewController != nil {
-            commentDetailViewController?.refreshView(comment: comment, notification: notification)
+        // Refresh existing content if possible
+        if let commentDetailViewController {
+            commentDetailViewController.refreshView(comment: comment, notification: notification)
         } else {
-            let commentDetailViewController = CommentDetailViewController(comment: comment,
-                                                                      notification: notification,
-                                                                      notificationDelegate: notificationDelegate,
-                                                                      managedObjectContext: managedObjectContext)
-
-            commentDetailViewController.view.translatesAutoresizingMaskIntoConstraints = false
-            add(commentDetailViewController)
-            view.pinSubviewToAllEdges(commentDetailViewController.view)
-            self.commentDetailViewController = commentDetailViewController
+            self.content?.viewController.remove()
+            let newContent = makeNewContent(with: comment)
+            let viewController = newContent.viewController
+            viewController.view.translatesAutoresizingMaskIntoConstraints = false
+            self.add(viewController)
+            self.view.pinSubviewToAllEdges(viewController.view)
+            self.content = newContent
         }
 
-        configureNavBarButtons()
+        // Configure navigation bar buttons
```

Comment:
Nit: This comment looks redundant as the function name is quite clear.

Reply:
Resolved in 67958ebd6eeb96f9c7deb5771baed43aaf0bc2aa